stablehlo/reference/ProcessGrid.h

```diff
@@ -140,6 +140,11 @@ class ProcessGrid {
                               ProcessId processId, const Tensor &operand);
 
  private:
+  /// Synchronizes access to `channelLocks_` to prevent multiple processes from
+  /// creating multiple mutexes when the entry in `channelLocks_` is empty.
+  std::mutex &getChannelsMasterLock(ProcessGroup processGroup,
+                                    ChannelId channelId);
+
   /// StableHLO `num_replicas`.
   uint32_t numReplicas_;
```

Comment:
While we're making changes - any thoughts on making `numReplicas_` and `numPartitions_` const? These are both read in ProcessGrid methods without any locking, so would be safer to enforce that these values are read-only.

Reply:
Agreed, updated.